pkg/testing/fetcher.go

```diff
@@ -85,18 +90,59 @@ type fetcherCache struct {
 func (c *fetcherCache) fetch(ctx context.Context, l Logger, res FetcherResult) (string, error) {
 	name := res.Name()
 	src := filepath.Join(c.dir, name)
-	_, err := os.Stat(src)
-	if err == nil || os.IsExist(err) {
-		l.Logf("Using existing artifact %s", name)
-		return src, nil
+
+	if c.isSameArtefact(ctx, src, res, l) {
+		_, err := os.Stat(src)
+		if err == nil || os.IsExist(err) {
+			l.Logf("Using existing artifact %s", name)
+			return src, nil
+		}
 	}
-	err = res.Fetch(ctx, l, c.dir)
+
+	err := res.Fetch(ctx, l, c.dir)
 	if err != nil {
 		return "", err
 	}
 	return src, nil
 }
 
+func (c *fetcherCache) isSameArtefact(ctx context.Context, path string, res FetcherResult, l Logger) bool {
```

Comment:
nit: Artefact vs Artifact :-)
There are 500+ "Artifact" and only 1 "Artefact" in the agent repo. Maybe keep it consistent.